### PR TITLE
Implement 2-hop loop avoidance.

### LINKED ISSUE #1086: Two-hop loops detection
I have experimented with large Thread network with OpenThread's nodes that send quite a big amount of application packets. Once in a while i saw loops (A->B->C->B->C->B->C->... ended when mesh hop limit expired). I did some quick code analysis and it looks like there is no two-hop loops detection as per 5.9.6.

### PR DESCRIPTION
**Pull request by jwhui**
Fixes #1086.

### CODE REVIEW

File: src/core/thread/mle_router.cpp [hunk 1]
```diff
@@ -3530,6 +3530,19 @@ ThreadError MleRouter::GetNextNeighborInfo(otNeighborInfoIterator &aIterator, ot
...
+void MleRouter::ResolveRoutingLoops(uint16_t aSourceMac, uint16_t aDestRloc16)
+{
+    if (aSourceMac == GetNextHop(aDestRloc16))
+    {
+        // loop detected
+        Router *router = GetRouter(GetRouterId(aDestRloc16));
+        assert(router != NULL);
+
+        // invalidate next hop
+        router->mNextHop = kInvalidRouterId;
```
Comment (LuDuda) [inline src/core/thread/mle_router.cpp]: I might be wrong but I believe we should reset the trickle timer here as well, as 3rd paragraph of 5.9.7 describes.

Comment (jwhui) [inline src/core/thread/mle_router.cpp]: Good catch.  Updated this PR.

**Comment by jwhui** [comment]
@nibanks, I believe the Thread Certification Test Plan does not explicitly test the 2-hop loop scenario - we'll have to make our own.

@LuDuda, do you have a scenario that could be added to the python tests?

**Comment by LuDuda** [comment]
I will try to figure out the scenario and i will add python test for it. However it probably will happen after the christmas.

**Pull request merged**